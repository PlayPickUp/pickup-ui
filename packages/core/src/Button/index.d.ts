import React from "react";
import { ButtonProps } from "../../../../types";
declare const Button: React.FC<
	ButtonProps & React.HTMLAttributes<HTMLOrSVGElement>
>;
export default Button;
//# sourceMappingURL=index.d.ts.map
