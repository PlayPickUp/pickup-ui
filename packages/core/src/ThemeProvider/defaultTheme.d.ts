import { DefaultTheme } from "../../../../types";
export declare const defaultTheme: DefaultTheme;
//# sourceMappingURL=defaultTheme.d.ts.map
