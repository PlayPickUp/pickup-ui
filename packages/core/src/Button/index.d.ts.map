{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["index.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAC;AAK1B,OAAO,EAAE,WAAW,EAAgB,MAAM,mBAAmB,CAAC;AAuB9D,QAAA,MAAM,MAAM,EAAE,KAAK,CAAC,EAAE,CACrB,WAAW,GAAG,KAAK,CAAC,cAAc,CAAC,gBAAgB,CAAC,CAwCpD,CAAC;AAEF,eAAe,MAAM,CAAC"}