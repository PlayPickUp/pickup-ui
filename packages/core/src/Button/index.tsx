import React from "react";
import { createUseStyles } from "react-jss";
import { Link } from "react-router-dom";
import classNames from "classnames";

import { ButtonProps, DefaultTheme } from "../types";

const useStyles = createUseStyles((theme: DefaultTheme) => ({
  root: {
    display: "flex",
    position: "relative",
    flexFlow: "row nowrap",
    justifyContent: "center",
    alignItems: "center",
    minHeight: 47,
    height: "auto",
    width: (props: ButtonProps) => (props.variant === "fit" ? "auto" : "100%"),
    padding: "0 30px",
    appearance: "none",
    borderRadius: theme.borderRadius,
    border: "none",
    background: theme.gradients.main,
    fontFamily: theme.typography.fontFamilies.headline,
    fontSize: 18,
    lineHeight: "24px",
    letterSpcing: "0.3px",
    color: theme.colors.white,
    textTransform: "uppercase",
    transition: "opacity 200ms ease-in-out",
    opacity: 1,
    fallbacks: {
      background: theme.colors.primary.base,
    },
    "&:hover": {
      opacity: 0.85,
    },
  },
}));

const Button: React.FC<
  ButtonProps & React.HTMLAttributes<HTMLOrSVGElement | HTMLInputElement>
> = ({
  className,
  children,
  disabled,
  element: Element = "button",
  href,
  onClick,
  style,
  to,
  variant = "fit",
  useSubmit,
  submitText,
  ...rest
}) => {
  const classes = useStyles({ variant });

  if (to) {
    return (
      <Link
        data-testid="pickup-button"
        to={to}
        className={classNames(classes.root, className)}
        style={style}
        onClick={onClick}
        {...rest}
      >
        {children}
      </Link>
    );
  }

  if (useSubmit) {
    if (!submitText) {
      console.warn("Must use submitText when useSubmit is set to true!");
    }
    return (
      <input
        data-testid="pickup-button"
        className={classNames(classes.root, className)}
        type="submit"
        value={submitText || "Submit"}
        onClick={onClick}
        disabled={disabled}
        style={style}
        {...rest}
      />
    );
  }

  return (
    <Element
      data-testid="pickup-button"
      className={classNames(classes.root, className)}
      disabled={disabled}
      style={style}
      onClick={onClick}
      href={href}
      {...rest}
    >
      {children}
    </Element>
  );
};

export default Button;
