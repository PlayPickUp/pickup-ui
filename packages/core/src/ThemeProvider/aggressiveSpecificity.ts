/* This is based on the unsupported jss-increase-specificity plugin at https://github.com/iamstarkov/jss-increase-specificity
JSS's original createGenerateId has been adapted here to use logic from that plugin. /*

// From the original plugin:
/**
 * use :not(#\20), :not(.\20) and :not(\20) instead of generating unlikely
 * appearing ids…
 * — twitter.com/subzey/status/829050478721896448
 * Rationale: \20 is a css escape for U+0020 Space, and neither classname,
 * nor id, nor tagname can contain a space
 * — twitter.com/subzey/status/829051085885153280
 */

import { Rule, CreateGenerateId } from "jss";
import { JssStyleSheet } from "../types";

var selector = ":not(#\\20)";
const repeat = 3;
const maxRules = 1e10;

export const increaseSpecificity: CreateGenerateId = (options = {}) => {
  let ruleCounter = 0;
  console.log({ options });
  return (rule: Rule, sheet?: JssStyleSheet): string => {
    ruleCounter += 1;
    if (ruleCounter > maxRules) {
      console.warn(
        false,
        `[JSS] You might have a memory leak. Rule counter is at ${ruleCounter}.`
      );
    }
    const jssId = String(sheet.options.jss.id) ?? "";
    const prefix = sheet.options.classNamePrefix ?? "";

    let suffix = "";
    for (let i = 0; i < repeat; i++) {
      suffix += selector;
    }

    return `${prefix + rule.key}${
      jssId ? `-${jssId}` : ""
    }-${ruleCounter}${suffix}`;
  };
};
