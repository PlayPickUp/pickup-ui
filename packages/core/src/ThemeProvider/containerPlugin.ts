/*
 *  Adds container params to all selectors
 *  Example: 'h1' becomes 'body #PickUpUI h1'
 */

import { Plugin } from "jss";

export function containUI(withReset: boolean, container?: string): Plugin {
  const prefix = container ? container : "body #PickUpUI ";

  // 'rule' is left untyped because the Rule import from jss is not recognizing all child properties
  function onProcessRule(rule): Plugin["onProcessRule"] {
    const parent = rule.options.parent;
    if (
      rule.type !== "style" ||
      rule.selectorText.includes(prefix) ||
      (parent && parent.type === "keyframes") ||
      (withReset && parent && parent.key === "@global")
    )
      return;

    if (rule.selectorText.includes(",")) {
      const allSelectors = rule.selectorText.split(",");
      const transformedSelectors = allSelectors.map((selector: string) =>
        selector.includes(prefix) ? selector : prefix + selector
      );
      rule.selectorText = transformedSelectors.join(",");
    } else {
      rule.selectorText = prefix + rule.selectorText;
    }
  }
  return { onProcessRule };
}
