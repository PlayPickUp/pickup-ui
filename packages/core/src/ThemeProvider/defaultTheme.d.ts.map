{"version":3,"file":"defaultTheme.d.ts","sourceRoot":"","sources":["defaultTheme.ts"],"names":[],"mappings":"AAAA,OAAO,EAMN,YAAY,EAWZ,MAAM,mBAAmB,CAAC;AAqJ3B,eAAO,MAAM,YAAY,EAAE,YAU1B,CAAC"}