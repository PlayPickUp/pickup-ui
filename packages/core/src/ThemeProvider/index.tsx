import React from "react";
import { jss, JssProvider, ThemeProvider as JSSThemeProvider } from "react-jss";
import { containUI } from "./containerPlugin";
import merge from "lodash/merge";

import { defaultTheme } from "./defaultTheme";

import { ThemeProviderProps } from "../types";
import GlobalsAndReset from "./GlobalsAndReset";

function generateRandom(digitCount) {
  return Math.floor(
      Math.pow(10, digitCount - 1) +
      Math.random() * 9 * Math.pow(10, digitCount - 1)
  );
}

const ThemeProvider: React.FC<ThemeProviderProps> = ({
  children,
  withReset = true,
  theme,
  ...rest
}) => {
  jss.use(containUI(withReset));
  const mergedTheme = theme ? merge(defaultTheme, theme) : defaultTheme;

  const random = generateRandom(5)

  return (
    <div id="PickUpUI">
      <JssProvider jss={jss} classNamePrefix={`PU${random}--`}>
        <JSSThemeProvider theme={mergedTheme} {...rest}>
          <GlobalsAndReset>{children}</GlobalsAndReset>
        </JSSThemeProvider>
      </JssProvider>
    </div>
  );
};

export default ThemeProvider;
