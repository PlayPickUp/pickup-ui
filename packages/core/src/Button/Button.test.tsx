import React from "react";
import { render } from "@testing-library/react";

import ThemeProvider from "../ThemeProvider";
import Button from "./index";

test("Renders and matches snapshot", () => {
  const { container } = render(
    <ThemeProvider>
      <Button href="https://www.playpickup.com">Make your pick</Button>
    </ThemeProvider>
  );
  expect(container).toMatchSnapshot();
});

test("Child text renders correctly", () => {
  const { getByText } = render(
    <ThemeProvider>
      <Button href="https://www.playpickup.com">Make your pick</Button>
    </ThemeProvider>
  );
  expect(getByText("Make your pick")).toBeTruthy();
});

test("Href is passed correctly", () => {
  const { getByTestId } = render(
    <ThemeProvider>
      <Button href="https://www.playpickup.com">Make your pick</Button>
    </ThemeProvider>
  );
  expect(getByTestId("pickup-button").getAttribute("href")).toEqual(
    "https://www.playpickup.com"
  );
});

test("Button is disabled when prop disabled is true", () => {
  const { getByTestId } = render(
    <ThemeProvider>
      <Button disabled href="https://www.playpickup.com">
        Make your pick
      </Button>
    </ThemeProvider>
  );
  expect(getByTestId("pickup-button").hasAttribute("disabled"));
});
