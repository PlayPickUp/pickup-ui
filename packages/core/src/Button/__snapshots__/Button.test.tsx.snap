// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders and matches snapshot 1`] = `
<div>
  <button
    class="PU--root-0-2-1 PU--root-d0-0-2-5"
    data-testid="pickup-button"
    href="https://www.playpickup.com"
  >
    Make your pick
  </button>
</div>
`;
